### DISCUSSION TITLE
Does Elegant work with firebase app hosting?

### DISCUSSION THREAD

**Discussion by Blackkadder** [elegantframework/elegant-cli#549] [Q&A] [answered]
Does Elegant work with firebase app hosting? We are trying to add Elegant to an existing nextjs project. Everything seems to work locally, but github login doesn't redirect when deployed in prod to firebase. 

Is this a known issue? Any help would be very appreciated. Thanks!

**Comment by Bowens20832** [comment]
Hey @Blackkadder Thank you for the question!

I have not tested Firebase personally yet, but I can add it to the list of hosts to ensure 100% compatibility with.

Did you generate a separate set of GitHub auth credentials for your production app?

If the app is redirecting back to localhost on login, then that means you need to create a duplicate GitHub app and update the URLs to reflect your production urls.

**Comment by Bowens20832** [comment]
Also, feel free to contact us on [Discord ](https://discord.com/channels/1115406043766329344/1194389666573594636)for immediate help : )

**Comment by Bowens20832** [comment]
Hey @Blackkadder, just following up 😃 

How did things go?

Did you get this issue resolved?

**Reply by Blackkadder** [reply, marked as answer]
The issue was with firebase overriding session cookies. Since our web deployment was pretty new, we moved it to a different hosting platform and it's now working as expected! Thanks!

**Reply by Bowens20832** [reply]
That is great news to hear! 

I hadn't considered Firebase when I started on the project, but I will be sure to test and ensure compatibility in the upcoming major version release.

Please let me know if you have any more questions or feedback. 

Thanks again for opening up the discussion 😎
